clients/src/main/java/org/apache/kafka/common/utils/AppInfoParser.java

```diff
@@ -57,7 +57,7 @@ public static void registerAppInfo(String prefix, String id) {
             AppInfo mBean = new AppInfo();
             ManagementFactory.getPlatformMBeanServer().registerMBean(mBean, name);
         } catch (JMException e) {
-            log.warn("Error registering AppInfo mbean", e);
+            throw new KafkaException(e);
```

Comment:
Seems like we should still include `Error registering AppInfo mbean` as a message giving more context. Same for the other case.